packages/gatsby/src/utils/api-node-helpers-docs.js

```diff
@@ -0,0 +1,293 @@
+
+/** */
+const GatsbyReporter = {
+  /**
+   * @callback GatsbyReporterFn
+   * @param {string} message Message to display
+   * @returns {void}
+   */
+
+  /**
+   * @callback GatsbyReporterFnWithError
+   * @param {string} message Message to display
+   * @param {Error}[error] Optional error object
+   * @returns {void}
+   */
+
+  /**
+   * @type {GatsbyReporterFn}
+   * @example
+   * reporter.info(`text`)
+   */
+  info: true,
+
+  /**
+   * @type {GatsbyReporterFn}
+   * @example
+   * reporter.warn(`text`)
+   */
+  warn: true,
+
+  /**
+   * @type {GatsbyReporterFnWithError}
+   * @example
+   * reporter.error(`text`, new Error('something'))
+   */
+  error: true,
+
+  /**
+   * @type {GatsbyReporterFnWithError}
+   * @example
+   * reporter.panic(`text`, new Error('something'))
+   */
+  panic: true,
+
+  /**
+   * @type {GatsbyReporterFnWithError}
+   * @example
+   * reporter.panicOnBuild(`text`, new Error('something'))
+   */
+  panicOnBuild: true,
+};
+
+/** */
+const GatsbyCache = {
+  /**
+   * @param {string} key
+   * @returns {Promise<any>}
+   * @example
+   * const value = await cache.get(`unique-key`)
+   */
+  get: true,
+
+  /**
+   * @param {string} key
+   * @param {any} value
+   * @returns {Promise<any>}
+   * @example
+   * await cache.set(`unique-key`, value)
+   */
+  set: true,
+};
+
+/** */
+const GatsbyTracing = {
+  /**
+   * Global tracer instance. Check
+   * [opentracing Tracer documentation](https://opentracing-javascript.surge.sh/classes/tracer.html)
+   * for more details.
+   * @type {Opentracing.Tracer}
+   */
+  tracer: true,
+
+  /**
+   * Tracer span representing API run. Check
+   * [opentracing Span documentation](https://opentracing-javascript.surge.sh/classes/span.html)
+   * for more details.
+   * @type {Opentracing.Span}
+   */
+  parentSpan: true,
+
+  /**
+   * @callback GatsbyTracing.StartSpan
+   * @param {string} spanName name of the span
+   * @returns {Opentracing.Span}
+   */
+
+  /**
+   * Convience function to start span using API run span as parent.
+   * @type {GatsbyTracing.StartSpan}
+   * @example
+   * exports.sourceNodes = async ({ actions, tracing }) => {
+   *   const span = tracing.startSpan(`foo`)
+   *
+   *   // Perform any span operations. E.g add a tag to your span
+   *   span.setTag(`bar`, `baz`)
+   *
+   *   // Rest of your plugin code
+   *
+   *   span.finish()
+   * }
+   */
+  startSpan: true,
+};
+
+/** */
+const GatsbyNodeHelpers = {
+  /**
+   * Key-value store used to persist results of time/memory/cpu instensive
+   * tasks. All functions are async and return promises.
+   * @type {GatsbyCache}
+   */
+  cache: true,
+
+  /**
+   * Get cache instance by name - this should only be used by plugins that
+   * accept subplugins.
+   * @param {string} id Test
+   * @returns {GatsbyCache} See [`cache`](#cache) section for reference.
+   */
+  getCache: true,
+
+  /**
+   * Create stable content digest from string or object. Useful to set
```

Comment:
How about something like this?

_Create a stable content digest from a string or object, you can use this to set the `internal.contentDigest` field on nodes. Gatsby uses the value of this field to invalidate stale data when your content changes._

Reply:
Reworded in https://github.com/gatsbyjs/gatsby/pull/12087/commits/b4f18e0730dd92a144a8a158c64513d00d1dcda8